### PR TITLE
Add a dropdown for selecting doc versions

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

https://user-images.githubusercontent.com/17039389/193045591-8db3f0e0-5238-4c4e-83f0-050f3689e65c.mov

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Manually tested.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
We need to backport the changes in this PR to the existing documents.

**Comment by harupy** [comment]
Backported and tested:

https://user-images.githubusercontent.com/17039389/193175865-55dbbdd7-ded1-478f-81c1-dbde5398c4fb.mov

File: docs/theme/mlflow/static/js/version-dropdown.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    });
+
+    var latestVersion = latestMicroVersions[0];
+    var docRegex = /\/docs\/(?<version>[^/]+)\//;
+    var currentVersion = docRegex.exec(window.location.pathname).groups.version;
+    var dropDown = document.createElement('select');
+    dropDown.onchange = function () {
+      var newHref = window.location.href.replace(docRegex, `/docs/${this.value}/`);
+      window.location.href = newHref;
```
Comment (BenWilson2) [inline docs/theme/mlflow/static/js/version-dropdown.js:40]:
```suggestion
      window.location.assign(docRegex, `/docs/${this.value}/`);
```
Would this enable using the back button / history? Doesn't a `.replace()` overwrite the current page in browser history?
I checked a few other python docs sites that have this feature of selecting versioning and they all have navigation persisted in the browser history.

Comment (BenWilson2) [reply docs/theme/mlflow/static/js/version-dropdown.js]: That comparison chart is so much nicer than searching through the react docs :)

**Review by BenWilson2 — APPROVED** [review]
LGTM! This is great!

**Pull request merged**